### DISCUSSION TITLE
Can IDE0130 be configured for another convention?

### DISCUSSION THREAD

**Discussion by Dryadxon** [dotnet/roslyn#58464] [Q&A] [answered]
Hi everyone,
As the title says, I would like to know if IDE0130 can be configured to also support another convention for the `namespace`s used.
My projects require the solution name to be part of the `namespace`, so if a class is in `PathToSolution\SolutionName\ProjectName\FolderName\ClassName.cs` the `namespace` I want to enforce is `SolutionName.ProjectName.FolderName`, instead of the `ProjectName.FolderName` suggested by IDE0130.
I'd like to use the rule, since it's very useful for organizing the code base, instead of simply disabling it, or making an analyzer just for this small convention difference.
Obviously, if this functionality is not feasible, or if it is not desirable to support multiple (or configurable) conventions, it is understandable.
Thank you for your time.

**Comment by sharwell** [comment, marked as answer]
The suggested namespace is based off the `RootNamespace` in the project, which can be configured in project properties:

![image](https://user-images.githubusercontent.com/1408396/147128433-3e1fed0e-98b4-4493-a2f5-edb4618ac0ae.png)

By default, this value does not include the solution name, but it can be customized to include it.

**Reply by Dryadxon** [reply]
Thank you so much for your answer! When dotnet/docs#23472 is resolved, it may be useful to add this information.
